File: problems/111/solution.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int minDepth(TreeNode* root) {
        if (!root) {
            return 0;
        }
        
        int ans = 0;
        queue<TreeNode*> q;
        q.push(root);
        
        while (!q.empty()) {
            ans++;
            
            for (int i = q.size(); i > 0; i--) {
                TreeNode* cur = q.front();
                q.pop();
                
                if (!cur->left && !cur->right) {
                    return ans;
                }
                
                if (cur->left) {
                    q.push(cur->left);
                }
                
                if (cur->right) {
                    q.push(cur->right);
                }
            }
        }
        
        return ans;
    }
};

File: problems/111/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def minDepth(self, root: TreeNode) -> int:
        if not root:
            return 0

        q, depth = deque([root]), 0
        
        while q:
            depth += 1
            
            for _ in range(len(q)):
                cur = q.popleft()
    
                if not cur.left and not cur.right:
                    return depth

                if cur.left:
                    q.append(cur.left)

                if cur.right:
                    q.append(cur.right)
        
        return 0